JSDemos/Demos/DataGrid/StateManagement/Angular/app/app.component.ts

```diff
@@ -0,0 +1,83 @@
+import { NgModule, Component, OnInit, enableProdMode } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { HttpClientModule } from '@angular/common/http';
+
+import { DxDataGridModule, DxLoadPanelModule } from 'devextreme-angular';
+import { Service, Order, Change } from './app.service';
+import { Observable } from 'rxjs';
+
+if (!/localhost/.test(document.location.host)) {
+    enableProdMode();
+}
+
+@Component({
+    selector: 'demo-app',
+    templateUrl: 'app/app.component.html',
+    styleUrls: ['app/app.component.css'],
+    providers: [Service],
+    preserveWhitespaces: true
+})
+export class AppComponent implements OnInit {
+    orders$: Observable<Order[]>;
+    changes: Change<Order>[] = [];
+    editRowKey?: number = null;
+    changesText = '';
+    isLoading = false;
+    loadPanelPosition = { of: '#gridContainer' };
+
+    constructor(private service: Service) { }
+
+    ngOnInit() {
+        this.orders$ = this.service.getOrders();
+
+        this.isLoading = true;
+        this.orders$.subscribe(() => {
+            this.isLoading = false;
+        });
+    }
+
+    onChangesChange(changes: Change<Order>[]) {
+        this.changesText = JSON.stringify(changes.map((change) => ({
+            type: change.type,
+            key: change.type !== 'insert' ? change.key : undefined,
+            data: change.data
+        })), null, ' ');
+    }
+
+    onSaving(e: any) {
+        const change = e.changes[0];
+
+        e.cancel = true;
```

Comment:
Move `e.cancel = true` to `if(change) {` block because save button should work if change does not exist.